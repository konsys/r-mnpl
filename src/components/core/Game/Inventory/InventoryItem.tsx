import React from "react";

export enum ItemLevel {
  ECONOM = 1,
  STANDARD = 2,
  SPECIAL = 3,
  HIGHEST = 4,
  EXCLUSIVE = 5,
}

export interface IInventoryItem {
  img: string;
  name: string;
  level: ItemLevel;
}

export default function InventoryItem({ img, name, level }: IInventoryItem) {
  return (
    <>
      <div className="item">
        <div className="itemImage">
          <div
            style={{
              backgroundImage: `url(${img})`,
            }}
          ></div>
        </div>
        <div className={`itemAbout level_${level}`}>
          <div>{name}</div>
        </div>
        <div className="itemMarks">
          <div className="itemMarksUsed"></div>
        </div>
      </div>
    </>
  );
}
