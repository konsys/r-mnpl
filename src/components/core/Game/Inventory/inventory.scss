@import "../../../../theme/styles/vars.scss";
.inventoryHeader {
  font-size: 28px;
  a {
    text-decoration: none;
  }
  .avatar {
    height: 40px;
    img {
      height: 100%;
      border-radius: 50%;
    }
  }
  .profileName {
    color: $shuffleСolor;
    font-size: 26px;
    border-bottom: 1px dotted $shuffleСolor;
  }
}
.inventoryFilters {
  color: red;
}

.inventoryItems {
  .item {
    position: relative;
    float: left;
    margin: 5.833333333333333px;
    cursor: pointer;
    width: 150px;

    .itemImage {
      width: 150px;
      height: 110px;
      padding: 14px 19px;
      border: 1px solid silver;
      border-bottom: none;
      border-radius: 5px 5px 0 0;
      > div {
        width: 100%;
        height: 100%;
        background: center/contain no-repeat;
      }
    }
    .itemAbout {
      &.level_1 {
        background-color: $itemLevel1;
      }
      &.level_2 {
        background-color: $itemLevel2;
      }
      &.level_3 {
        background-color: $itemLevel3;
      }
      &.level_4 {
        background-color: $itemLevel3;
      }
      &.level_5 {
        background-color: $itemLevel4;
      }
      border-radius: 0 0 5px 5px;

      display: table;
      width: 150px;
      height: 40px;
      padding: 3px 10px;
      line-height: 17px;
      font-size: 13px;
      color: #fff;
      > div {
        display: table-cell;
        vertical-align: middle;
      }
    }
    .itemMarks {
      position: absolute;
      top: 0;
      left: 0;
      padding: 3px;
      width: 100%;
      height: 25px;
      > div {
        float: right;
        width: 25px;
        height: 25px;
        font-family: Ionicons;
        line-height: 25px;
        text-align: center;
      }
      .itemMarksUsed {
        font-size: 25px;
        color: #8cc152;
        &::before {
          content: "\f3ff";
        }
      }
    }
  }
}
