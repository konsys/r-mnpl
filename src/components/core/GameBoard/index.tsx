import React from "react";

import { useStore } from "effector-react";

import { IFieldModalPosition } from "../../../types/types";
import { GameBoardView } from "./GameBoardView";

export const fieldsActionPosition = (): IFieldModalPosition => {
  return {
    left: 0,
    top: 0,
  };
};

export const GameBoard = () => {
  return <GameBoardView />;
};
