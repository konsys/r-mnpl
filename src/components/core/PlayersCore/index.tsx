import React from "react";
import { RouteComponentProps } from "@reach/router";

interface Props extends RouteComponentProps {}

//TODO unused now, is it needed?
export const Game = (props: Props) => {
  return <>GAME</>;
};
