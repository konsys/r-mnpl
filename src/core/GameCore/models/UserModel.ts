import { GameDomain } from "./BoardStore";
import { IUser } from "../../../components/Players/Players";

const UserDomain = GameDomain.domain("UserDomain");
export const resetUserEvent = UserDomain.event();

const init: IUser = {
  userId: 1,
  name: "Konstantin",
  vip: true,
  isActive: true,
  isBlocked: false
};

export const setUserEvent = UserDomain.event<IUser>();

export const modalStore = UserDomain.store(init)
  .on(setUserEvent, (_, data) => {
    console.log("setUserEvent", data);
  })
  .reset(resetUserEvent);
