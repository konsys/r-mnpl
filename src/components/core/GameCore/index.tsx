import React from "react";
import { RouteComponentProps } from "@reach/router";
import "./style.scss";
import { Chat } from "../../views/Chat/Chat";
import { Contract } from "../../views/Contract/Contract";
import { TableHelper } from "../../views/TableHelper/TableHelper";
import { Tokens } from "../../views/Tokens/Tokens";
import { Ticket } from "../../views/Ticket/ticket";
import { Arbitr } from "../../views/Arbitr/Arbitr";
import { M1tv } from "../../views/M1tv/M1tv";
import { BoardModal } from "../../views/BoardModal/BoardModal";
import { Dices } from "../../views/Dices/Dices";
import { GameLoading } from "../../views/GameLoading/GameLoading";
import { useStore } from "effector-react";
import { BoardCore } from "../BoardCore/BoardCore";
import { UsersCore } from "../PlayersCore/PlayersCore";
import { actionsStore } from "../../../stores/ActionStore";
import { IncomeMessageType } from "../../../types/BoardTypes";
import "react-toastify/dist/ReactToastify.css";
import { ToastContainer } from "react-toastify";

interface Props extends RouteComponentProps {}

export const Game = (props: Props) => {
  const actionState = useStore(actionsStore);
  const actionType = actionState.event.action.type;

  const isModalAction =
    actionType === IncomeMessageType.INCOME_AUCTION_MODAL ||
    actionType === IncomeMessageType.INCOME_CAN_BUY_MODAL ||
    actionType === IncomeMessageType.INCOME_ROLL_DICES_MODAL ||
    actionType === IncomeMessageType.INCOME_TAX_PAYING_MODAL ||
    actionType === IncomeMessageType.INCOME_UN_JAIL_MODAL;

  return (
    <>
      <div className="wrapper" style={{ width: "100%", height: "100%" }}>
        <div className="table _shakehack">
          <div className="table-body">
            <UsersCore />
            <div className="table-body-board">
              <BoardCore />
              <div className="table-body-board-center">
                <M1tv />
                {actionState && (
                  // TODO check for user credentials
                  // userState.userId === actionState.event.action.userId &&
                  <BoardModal isModal={isModalAction} />
                )}
                <Arbitr />
                <Ticket />
                <Chat />
              </div>
              <div className="table-body-board-tokens">
                <Tokens />
              </div>
              <Dices />
              <Contract />
              <TableHelper />
            </div>
          </div>

          <div id="placeholder_gameover"></div>
        </div>
        <div className="table-jokes"></div>
      </div>
      <GameLoading />
      <ToastContainer />
    </>
  );
};
