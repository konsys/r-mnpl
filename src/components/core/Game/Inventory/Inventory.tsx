import "./inventory.scss";

import { Button, Grid, Select, TextField, Typography } from "@material-ui/core";
import InventoryItem, { ItemLevel } from "./InventoryItem";
import React, { useState } from "react";

import Alert from "@material-ui/lab/Alert";
import { InventoryType } from "types/types";
import { Link } from "react-router-dom";
import { Params } from "config/params";
import Template from "components/views/Template/Template";
import { useStore } from "effector-react";
import { useTranslation } from "react-i18next";
import { user$ } from "stores/Game/User/UserModel";

export default function Inventory() {
  const user = useStore(user$);
  const { t } = useTranslation();
  const [inventory, setInventory] = useState<string>("all");
  return (
    <>
      <Template columns={1} title={`${t("Inventory")} ${user?.name || ""}`}>
        {!user ? (
          <Grid container justify="center" alignContent="stretch">
            <Alert severity="error">
              {t("Login to manage your inventory")}
            </Alert>
          </Grid>
        ) : (
          <Grid
            container
            direction="column"
            spacing={5}
            className={"inventory"}
          >
            <Grid item style={{ border: "3px solid blue" }}>
              <Grid
                container
                alignItems="center"
                justify="space-between"
                direction="row"
                className={"inventoryHeader"}
              >
                <Grid item>
                  <Grid container spacing={1} alignItems="center">
                    <Grid item>
                      <div className="avatar">
                        {user?.avatar && (
                          <Link to={`/profile/${user.userId}`}>
                            <img
                              alt={user.name}
                              src={Params.BASE_URL + "/" + user.avatar}
                            />
                          </Link>
                        )}
                      </div>
                    </Grid>
                    <Grid item>
                      <Link to={`/profile/${user?.userId}`}>
                        <Typography className="profileName">
                          {(user && user.name.toUpperCase()) || ""}
                        </Typography>
                      </Link>
                    </Grid>
                    <Grid item>
                      <Typography color="textSecondary" variant="h6">
                        {">"}
                      </Typography>
                    </Grid>
                    <Grid item>{t("Inventory")}</Grid>
                    <Grid item>
                      <Typography color="textSecondary">12</Typography>
                    </Grid>
                  </Grid>
                </Grid>
                <Grid item>
                  <Button variant="outlined" color="primary">
                    {t("Exchange offers")}
                  </Button>
                </Grid>
              </Grid>
            </Grid>
            <Grid item>
              <Grid
                container
                alignItems="center"
                justify="flex-start"
                direction="row"
                className={"inventoryFilters"}
              >
                <Grid item>
                  <TextField
                    style={{ minWidth: "330px" }}
                    variant="outlined"
                    label={t("Input words for items search")}
                  />
                </Grid>
                <Grid item>
                  <Select
                    variant="outlined"
                    native
                    value={inventory}
                    onChange={(e) => setInventory(InventoryType.CARDS)}
                    inputProps={{
                      name: "age",
                      id: "filled-age-native-simple",
                    }}
                  >
                    <option value={""}>{t("All items")}</option>
                    <option value={InventoryType.CARDS}>{t("Cards")}</option>
                    <option value={InventoryType.CASES}>
                      {t("Cases and sets")}
                    </option>
                    <option value={InventoryType.DICES}>
                      {t("Dices and generators")}
                    </option>
                    <option value={InventoryType.BADGES}>{t("Badges")}</option>
                    <option value={InventoryType.STICKERS}>
                      {t("Stickers")}
                    </option>
                    <option value={InventoryType.OTHERS}>{t("Others")}</option>
                  </Select>
                </Grid>
              </Grid>
            </Grid>
            <Grid item>
              <Grid container direction="row" className="inventoryItems">
                <Grid item>
                  {new Array(10).fill(0).map(() => (
                    <InventoryItem
                      img={
                        "https://cdn2.kirick.me/libs/monopoly/fields/brands/0_auto/kia.svg"
                      }
                      name={"KIA"}
                      level={ItemLevel.USUAL}
                    />
                  ))}
                </Grid>
              </Grid>
            </Grid>
          </Grid>
        )}
      </Template>
      ))
    </>
  );
}
