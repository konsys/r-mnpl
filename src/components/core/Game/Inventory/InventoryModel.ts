import { GameDomain, user$ } from "stores/Game/User/UserModel";
import { guard, merge, sample } from "effector";

import { IInventory } from "types/types";
import { createGate } from "effector-react";
import { inventoryFetch } from "api/Inventory/api";

export const InventoryDomain = GameDomain.domain("InventoryDomain");

export const InventoryGate = createGate<{ userId: number | null }>();

export const getInventoryFx = InventoryDomain.effect<number, IInventory, Error>(
  {
    handler: inventoryFetch,
  }
);

export const inventory$ = InventoryDomain.store<IInventory | null>(null)
  .on(getInventoryFx.done, (prev, { result }) => result)
  .reset(InventoryGate.close);

const inventorySample = sample({
  clock: merge([InventoryGate.open, user$]),
  source: user$,
  fn: (v) => v?.userId,
});

guard({
  source: inventorySample,
  filter: InventoryGate.status && user$.map((v) => !!v?.userId),
  target: getInventoryFx,
});
