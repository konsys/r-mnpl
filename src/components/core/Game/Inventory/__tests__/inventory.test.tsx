import { logout, setUserEvent } from "stores/Game/User/UserModel";
import { testAvatarUser, testUser } from "testMocks/user";

import Alert from "@material-ui/lab/Alert";
import { Grid } from "@material-ui/core";
import Inventory from "../Inventory";
import { Link } from "react-router-dom";
import React from "react";
import { shallow } from "enzyme";

describe("Buy gallery test", () => {
  it("should render Inventory", () => {
    logout();
    expect(shallow(<Inventory />)).toMatchSnapshot();
  });

  it("should render Inventory", () => {
    setUserEvent(testUser);
    expect(shallow(<Inventory />)).toMatchSnapshot();
  });

  it("should show not logged page", () => {
    logout();
    const wrap = shallow(<Inventory />);
    expect(wrap.find(Alert)).toHaveLength(1);
    expect(wrap.find(Grid)).toHaveLength(1);
    expect(wrap.find(Alert).text()).toBe("Login to manage your inventory");
  });

  it("should show logged page", () => {
    setUserEvent(testUser);
    const wrap = shallow(<Inventory />);
    expect(wrap.find(Alert)).toHaveLength(0);
    expect(wrap.find(Grid)).toHaveLength(18);
  });

  it("should show logged page", () => {
    setUserEvent(testUser);
    expect(shallow(<Inventory />).find(Link)).toHaveLength(1);
    setUserEvent(testAvatarUser);
    expect(shallow(<Inventory />).find(Link)).toHaveLength(2);
  });

  it("should show links", () => {
    setUserEvent(testAvatarUser);
    const wrap = shallow(<Inventory />);
    expect(wrap.find(Link).get(0).props.to).toBe(
      `/profile/${testAvatarUser.userId}`
    );
    expect(wrap.find(Link).get(1).props.to).toStrictEqual(
      `/profile/${testAvatarUser.userId}`
    );
  });
});
