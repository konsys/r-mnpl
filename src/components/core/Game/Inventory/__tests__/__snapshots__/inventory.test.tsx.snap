// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buy gallery test should render Inventory 1`] = `ShallowWrapper {}`;

exports[`Buy gallery test should render Inventory 2`] = `ShallowWrapper {}`;
